Assets/MRTK/Core/Utilities/WebRequestRest/Rest.cs

```diff
@@ -254,31 +254,35 @@ private static async Task<Response> ProcessRequestAsync(UnityWebRequest webReque
             webRequest.disposeCertificateHandlerOnDispose = disposeCertificateHandlerOnDispose;
             await webRequest.SendWebRequest();
 
+            long responseCode = webRequest.responseCode;
+            byte[] downloadHandlerData = webRequest.downloadHandler?.data;
+            string downloadHandlerText = await Task
+                .Run(() => System.Text.Encoding.Default.GetString(downloadHandlerData));
+
 #if UNITY_2020_1_OR_NEWER
             if (webRequest.result == UnityWebRequest.Result.ConnectionError || webRequest.result == UnityWebRequest.Result.ProtocolError)
 #else
             if (webRequest.isNetworkError || webRequest.isHttpError)
 #endif // UNITY_2020_1_OR_NEWER
             {
-                if (webRequest.responseCode == 401) { return new Response(false, "Invalid Credentials", null, webRequest.responseCode); }
+                if (webRequest.responseCode == 401) { return new Response(false, "Invalid Credentials", null, responseCode); }
```

Comment:
Can this reference the new cached value too?

```suggestion
                if (responseCode == 401) { return new Response(false, "Invalid Credentials", null, responseCode); }
```

Reply:
Yes that's great